### PR TITLE
DOC: make return type documentation of series methods consistent #35409

### LINKED ISSUE #35409: DOC: Inconsistent return type documentation for functions with "inplace" option
#### Location of the documentation

This proposal concerns all methods with the `inplace` keyword argument. Three examples:

1 (Return type described as "blah-blah-type or `None` if `inplace=True`) https://pandas.pydata.org/docs/dev/reference/api/pandas.CategoricalIndex.add_categories.html?highlight=inplace

2 (Return type specified, but doesn't mention `None`) https://pandas.pydata.org/docs/dev/reference/api/pandas.Series.interpolate.html?highlight=inplace

#### Documentation problem

Many methods in pandas have keyword argument `inplace`. When `inplace=True` is supplied, the method updates `self` and returns `None`. When `inplace=False` is supplied, the method does not affect `self` and returns a changed copy of `self`.

Some of these functions' return types are documented as "Returns blah-type if inplace=True". Others just say "Returns blah-type". This is confusing because it suggests inconsistencies in *behavior* that are not really there.

#### Suggested fix for documentation

Find all methods with the `inplace` keyword argument. Change the "returns" section of their docstrings to this form:
```
        Returns
        -------
        Series or DataFrame
            Returns the same object type as the caller, interpolated at
            some or all ``NaN`` values. (Or return `None` if
            `inplace=True`.)
```
Also change all the formal return type signatures to the form "--> Optional[Foo]" if they aren't all already like that.

### PR DESCRIPTION
**Pull request by junjunjunk**
- [x] closes #35409 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by junjunjunk** [comment]
Fix docstrings associated with Series methods for now.
Please review.

Once it passes review, I'll start working on other docstring fixes.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9048,9 +9048,9 @@ def where(
         try_cast : bool, default False
             Try to cast the result back to the input type (if possible).
 
-        Returns
+        Returns or None
```
Comment (ivanovmg) [inline pandas/core/generic.py:9051]: I guess that this ``or None`` is unnecessary.

Comment (junjunjunk) [reply pandas/core/generic.py:9051]: Oh, this is a mistake. Thank you.

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -917,7 +916,8 @@ def reorder_categories(self, new_categories, ordered=None, inplace=False):
 
         Returns
         -------
-        cat : Categorical with reordered categories or None if inplace.
+        cat : Categprocal or None
```
Comment (ivanovmg) [inline pandas/core/arrays/categorical.py:919]: Typo here. Otherwise, lgtm

Comment (junjunjunk) [reply pandas/core/arrays/categorical.py]: Thank you. Done.
 I start working on other docstring fixes in the same way.

Comment (ivanovmg) [reply pandas/core/arrays/categorical.py]: Note that CI Checks complains about missing dots at the end of return statement.
```
6818:RT05:pandas.DataFrame.interpolate:Return value description should finish with "."
```

Comment (junjunjunk) [reply pandas/core/arrays/categorical.py]: I'm sorry for all the mistakes. Thank you for pointing that out.

**Review by ivanovmg — APPROVED** [review]

**Comment by junjunjunk** [comment]
would love to get a hacktoberfest tag if possible. #36837

**Comment by jreback** [comment]
cc @simonjayhawkins ok with the text?

**Comment by simonjayhawkins** [comment]
Thanks @junjunjunk! Very nice.

**Pull request merged**